tests/test_sync.py

```diff
@@ -203,6 +204,21 @@ def once_handler(data):
     assert ee._events['event'] == OrderedDict()
 
 
+def test_once_remove_race():
```

Comment:
Oof, yeah, OK, I see what you mean with this test. I wish there was a sensible way to test this, but I think that given what we have here it's too hairy to implement what we had in mind. My vote is to remove this test and shrug.